packages/contracts-bedrock/contracts/test/CrossDomainMessenger.t.sol

```diff
@@ -17,3 +23,100 @@ contract CrossDomainMessenger_BaseGas_Test is Messenger_Initializer {
         L1Messenger.baseGas(hex"ff", _minGasLimit);
     }
 }
+
+// CrossDomainMessenger_RelayMessage_Test tests re-entrency of relayMessage.
+contract CrossDomainMessenger_RelayMessage_Test is Messenger_Initializer {
+    // Storage slot of the l2Sender
+    uint256 constant senderSlotIndex = 50;
+
+    address public fuzzedSender;
+    address public sender;
+
+    address public target;
+
+    // Internal helper function to relay a message and perform assertions.
+    function internalRelay(address innerSender) internal {
+        assertEq(sender, L1Messenger.xDomainMessageSender());
+
+        bytes memory callMessage = abi.encodeWithSelector(
+            CrossDomainMessenger_RelayMessage_Test.relayMessage.selector
+        );
+
+        bytes32 hash = Hashing.hashCrossDomainMessage(
```

Comment:
I would love it if you used kwargs style syntax here so we know what the 0s are

Reply:
Good call - will use kwargs going forward 👍  (maybe we should also consider putting in a gardening ticket or sub-ticket underneath a styleguide compliance ticket)